### PR TITLE
Refactored GalacticBankManager

### LINKED ISSUE #919: Refactor GalacticBankManager into an EntitySystem
I think `_components` can be replaced with the EntitySystem `RelevantEntities`.

### PR DESCRIPTION
**Pull request by DamianX**
Closes #919

I initially went into this with the intent of using RelevantEntities from EntitySystem to iterate over the CargoConsoleComponents but digging a little deeper I realized SyncComponents and SyncComponentsWithId can be replaced by a simple event OnBalanceChange that I have now defined on CargoBankAccount.

I made everything pretty and doc-commented, too.

### CODE REVIEW

File: Content.Server/GameObjects/Components/Cargo/CargoConsoleComponent.cs [hunk 1]
```diff
@@ -53,8 +75,9 @@ public override void Initialize()
             _userInterface = Owner.GetComponent<ServerUserInterfaceComponent>().GetBoundUserInterface(CargoConsoleUiKey.Key);
             _userInterface.OnReceiveMessage += UserInterfaceOnOnReceiveMessage;
             _powerDevice = Owner.GetComponent<PowerDeviceComponent>();
-            _galacticBankManager.AddComponent(this);
-            BankId = 0;
+            EntitySystem.TryGet(out _cargoConsoleSystem);
```
Comment (ComicIronic) [inline Content.Server/GameObjects/Components/Cargo/CargoConsoleComponent.cs:78]: Any reason for using `TryGet` over `Get` here?

File: Content.Server/GameObjects/Components/Cargo/CargoConsoleComponent.cs [hunk 2]
```diff
@@ -53,8 +75,9 @@ public override void Initialize()
...
+            BankAccount = _cargoConsoleSystem.StationAccount;
+            OnBankAccountBalanceChange(); // So that the UI isn't empty before the first update
```
Comment (ComicIronic) [inline Content.Server/GameObjects/Components/Cargo/CargoConsoleComponent.cs:80]: Would it make sense to move this to the `BankAccount` setter?

File: Content.Server/GameObjects/Components/Cargo/CargoConsoleComponent.cs (hunk 1 above)
Comment (DamianX) [reply Content.Server/GameObjects/Components/Cargo/CargoConsoleComponent.cs:78]: The reason is that I didn't know that existed, I'll change it now

File: Content.Server/GameObjects/Components/Cargo/CargoConsoleComponent.cs [hunk 3]
```diff
@@ -80,7 +103,7 @@ private void UserInterfaceOnOnReceiveMessage(ServerBoundUserInterfaceMessage ser
                 {
                     if (msg.Amount <= 0)
                         break;
-                    _cargoOrderDataManager.AddOrder(Orders.Database.Id, msg.Requester, msg.Reason, msg.ProductId, msg.Amount, BankId);
+                    _cargoOrderDataManager.AddOrder(Orders.Database.Id, msg.Requester, msg.Reason, msg.ProductId, msg.Amount, _cargoConsoleSystem.StationAccount.Id);
```
Comment (ComicIronic) [inline Content.Server/GameObjects/Components/Cargo/CargoConsoleComponent.cs:106]: Not `_bankAccount.Id`?

File: Content.Server/GameObjects/Components/Cargo/CargoConsoleComponent.cs [hunk 4]
```diff
@@ -96,7 +119,7 @@ private void UserInterfaceOnOnReceiveMessage(ServerBoundUserInterfaceMessage ser
                     _prototypeManager.TryIndex(order.ProductId, out CargoProductPrototype product);
                     if (product == null)
                         break;
-                    if (!_galacticBankManager.ChangeBalance(BankId, (-product.PointCost) * order.Amount))
+                    if (!_cargoConsoleSystem.ChangeBalance(_cargoConsoleSystem.StationAccount.Id, (-product.PointCost) * order.Amount))
```
Comment (ComicIronic) [inline Content.Server/GameObjects/Components/Cargo/CargoConsoleComponent.cs:122]: Same question.

File: Content.Server/GameObjects/Components/Cargo/CargoConsoleComponent.cs (hunk 3 above)
Comment (DamianX) [reply Content.Server/GameObjects/Components/Cargo/CargoConsoleComponent.cs:106]: Good catch, forgot to change these two

**Pull request merged**